resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,307 @@
+if ([string]::IsNullOrEmpty($env:TestRegistryPath)) {
+    $global:tzAutoUpdatePath = 'HKLM:\SYSTEM\CurrentControlSet\Services\W32Time\Parameters'
+    $global:timeZoneInformationPath = 'HKLM:\SYSTEM\CurrentControlSet\Control\TimeZoneInformation'
+} else {
+    $global:tzAutoUpdatePath = $global:timeZoneInformationPath = $env:TestRegistryPath
+}
+
+#region Functions
+function TryGetRegistryValue {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )
+
+    if (Test-Path -Path $Key) {
+        try {
+            return (Get-ItemProperty -Path $Key -Name $Property -ErrorAction SilentlyContinue | Select-Object -ExpandProperty $Property)
+        } catch {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    } else {
+        Write-Verbose 'Registry key does not exist.'
+    }
+}
+function Get-ValidTimeZone {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $TimeZone
+    )
+
+    try {
+        $timeZoneId = (Get-TimeZone -Id $TimeZone -ErrorAction Stop).Id
+    } catch {
+        throw [System.Configuration.ConfigurationException]::new("Executing 'Get-TimeZone' failed. Error: $($PSItem.Exception.Message)")
+    }
+
+    return $timeZoneId
+}
+#endRegion Functions
+
+#region Classes
+<#
+.SYNOPSIS
+    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+
+.DESCRIPTION
+    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+
+.PARAMETER TimeZone
+    The time zone to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
+
+.PARAMETER SetTimeZoneAutomatically
+    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
+
+.PARAMETER AdjustForDaylightSaving
+    Whether to adjust for daylight saving time. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Adjust for daylight saving time automatically.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
+
+    This example sets the time zone to Pacific Standard Time.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+
+    This example gets the current time settings on the machine.
+#>
+[DscResource()]
+class TimeZone {
+    [DscProperty(Key)]
+    [string] $TimeZone
+
+    [DscProperty()]
+    [nullable[bool]] $SetTimeZoneAutomatically
```

Comment:
I think a separate DSC for this field specifically. It doesn't makes sense to have to specify a timeZone (key) in order to be able to set this value. Also this value has no effect until you give it a manual time to set it to.

Reply:
After debugging for countless of hours, I got the confusing. Let me try to eleborate what is adjusted now. Imagine the following:

![image](https://github.com/user-attachments/assets/9281c330-0cc1-4a8c-95af-ecddad0fa61c)

Both settings related to time, so I can recon that it belongs to the `TimeZone` class. I have now updated both properties. The situation is now as followed:

1. When a user sets the **SetTimeZoneAutomatically** to `$true`, it overrules all others even when the user sets them
2. When a user leaves it empty and sets the other values, they are set correctly. Here I made a small adjustment for daylight savings, because there are just to many registry keys to be modified. I'm using _tzutil_ to adjust this.

I have tried to follow the principle that @denelon mentioned in #34 .

Please revisit the adjustments once again :)